02/ht/example/script.js

```diff
@@ -31,49 +52,104 @@ function bind(func, context) {
  * (можно использовать фукнцию выше)
  */
 
+Function.prototype.myBind = function (context) {
+  return bind(this, context);
+};
+
 /**
-* создать объект с волшебным свойством,
-* чтобы при присвоении ему значения, в консоль выводилась текущая дата и значение, которое присваиваем.
-* А при чтении всегда выводилось число на 1 больше предыдущего
-* o.magicProperty = 5; // 'Sat Mar 24 2018 13:48:47 GMT+0300 (+03) -- 5'
-* console.log(o.magicProperty); // 6
-* console.log(o.magicProperty); // 7
-* console.log(o.magicProperty); // 8
-*/
+ * создать объект с волшебным свойством,
+ * чтобы при присвоении ему значения, в консоль выводилась текущая дата и значение, которое присваиваем.
+ * А при чтении всегда выводилось число на 1 больше предыдущего
+ * o.magicProperty = 5; // 'Sat Mar 24 2018 13:48:47 GMT+0300 (+03) -- 5'
+ * console.log(o.magicProperty); // 6
+ * console.log(o.magicProperty); // 7
+ * console.log(o.magicProperty); // 8
+ */
+
+const o = {
+  count: 0,
+  get magicProperty() {
+    return (this.count += 1);
+  },
+  set magicProperty(value) {
+    this.count = value;
+    console.log(new Date() + "--" + this.count);
+  },
+};
 
 /**
-* Создать конструктор с методами, так,
-* чтобы следующий код работал и делал соответствующие вещи
-* те запуск кода ниже должен делать то, что говорят методы
-* u.askName().askAge().showAgeInConsole().showNameInAlert();
-*/
+ * Создать конструктор с методами, так,
+ * чтобы следующий код работал и делал соответствующие вещи
+ * те запуск кода ниже должен делать то, что говорят методы
+ * u.askName().askAge().showAgeInConsole().showNameInAlert();
+ */
+
+function ASK() {
+  this.askName = function () {
+    this.name = prompt("You name?", "");
+    return this;
+  };
+  this.askAge = function () {
+    this.age = prompt("Age?", "");
+    return this;
+  };
+  this.showAgeInConsole = function () {
+    console.log("You name is " + this.age);
+    return this;
+  };
+  this.showNameInAlert = function () {
+    alert("You name is " + this.name);
+    return this;
+  };
+}
 
 /**
  * Написать фукнцию-калькулятор, которая работает следующим образом
  * calculate('+')(1)(2); // 3
  * calculate('*')(2)(3); // 6
  * Допустимые операции : + - * /
  */
-function calculate() {
-  /* put your code here */
+
+function calculate(z) {
+  return function (a) {
+    return function (b) {
+      switch (z) {
+        case "+":
+          return a + b;
+        case "*":
+          return a * b;
+        case "-":
+          return a - b;
+        case "/":
+          return a / b;
+      }
+    };
+  };
 }
 
 /**
  * Создайте конструктор-синглтон? Что такое синглтон?
  * new Singleton() === new Singleton
  */
 function Singleton() {
-  throw 'undefined';
+  if (Singleton.example === undefined) {
```

Comment:
Буду ругаться за плохое именование. Старайся хоть минимально отображать смысл того, чем называешь.
```suggestion
  if (Singleton.self=== undefined) {
```